dspace-server-webapp/src/test/java/org/dspace/app/rest/AuthorizationRestRepositoryIT.java

```diff
@@ -2575,22 +2575,22 @@ public void findByMultipleObjectsAndFeatureInternalServerErrorTest() throws Exce
 
         // verify that it works for normal loggedin users - with eperson parameter
         getClient(epersonToken).perform(get("/api/authz/authorizations/search/objects")
-            .param("type", "core.site")
+            .param("type", "core.sites")
             .param("uuid", siteId)
             .param("feature", alwaysException.getName())
             .param("eperson", eperson.getID().toString()))
             .andExpect(status().isInternalServerError());
 
         // verify that it works for normal loggedin users - without eperson parameter
         getClient(epersonToken).perform(get("/api/authz/authorizations/search/objects")
-            .param("type", "core.site")
+            .param("type", "core.sites")
             .param("uuid", siteId)
             .param("feature", alwaysException.getName()))
             .andExpect(status().isInternalServerError());
 
         // verify that it works for anonymous users
         getClient().perform(get("/api/authz/authorizations/search/objects")
-            .param("type", "core.site")
+            .param("type", "core.sites")
```

Comment:
this is quite impactful as with the previous implementation essentially plural and singular were working equivalent. This is because if we apply the makeSingular function to singular word we get the original input and the lookup for the rest repositories was based on the singular name. With this PR we are looking up the rest repository by their plural name without performing transformation of the user input that mean that we have now a much strict control as demonstrate the need to fix all the url to use the right plural form in the test.
We need to highlight this change as a huge breaking change in DSpace 8.

Please note that also the URI example in https://github.com/DSpace/RestContract/blob/main/authorizations.md#object must be updated to use the correct plural form

Last but not least, due to the lack of e2e coverage for angular, I'm a bit worry about the absence of a corresponding angular PR... maybe our angular UI is already doing things in the right way always using the plural form but what about the `/api/authz/authorizations/search/objects?uuid=<:uuid>&type=<:type>[&eperson=<:uuid>&feature=<:string>]` endpoint? is it not used at all in angular @atarix83 ?

Reply:
@tdonohue: I already created the rest contract PR to fix the singular endpoints, but I did not create the `CHANGELOG.md` file yet since I don't know if this needs to be in a certain format/what kind of info should be added

Reply:
@alexandrevryghem : I'll take care of the CHANGELOG.md once this PR is ready.  You can leave this comment "unresolved" and I'll resolve it once it is documented. So, please just address everything else.